app/models/user.rb

```diff
@@ -19,8 +19,13 @@ class User < ApplicationRecord
   has_many :likes, dependent: :destroy
   has_many :liked_posts, through: :likes, source: 'post'
   has_many :friendships
+  has_many :received_but_unconfirmed, -> { where confirmed: false }, class_name: 'Friendship', foreign_key: 'friend_id'
+  has_many :pending_friends, through: :received_but_unconfirmed, source: 'user'
   has_many :inverted_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
```

Comment:
- [ ] `all_friends` method name indicates that it does return list of friend users, but it returns friendships, which is confusing. So I suggest to rafactor it to return friends from both `confirmed_friendships` and `inverse_confirmed_friendships`:
```ruby
# Your associations
has_many :confirmed_friendships, -> { where confirmed: true }, class_name: 'Friendship'
has_many :friends, through: :confirmed_friendships
# Added associations
has_many :inverse_confirmed_friendships, -> { where confirmed: true }, class_name: 'Friendship', foreign_key: 'friend_id'
has_many :friends, through: :inverse_confirmed_friendships, source: 'user'
```
And to get all friends => `friends + inverse_friends`